File: src/PizzaPlus/Pizza.java
package PizzaPlus;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class Pizza implements Serializable {

    ArrayList<Toppings> toppings = new ArrayList<>();
    private double pizzaTotal;
    Customer customer;
    Toppings[] currToppings;
    double basePrice;
    XmlReader toppingsXml = new XmlReader("toppings.xml");

    static double pie12 = 12.00, pie14 = 14.00, pie16 = 16.00;


    public Pizza(double basePrice){
        this.basePrice = basePrice;
    }
    public void toppingsScroller() throws FileNotFoundException {
//        currToppings = toppingsXml.getToppingsList();

        List<Toppings> users = new Gson().fromJson(new FileReader("toppings.json"),new TypeToken<List<Toppings>>() {}.getType());
        users.forEach(System.out::println);
//        for (int i = 0; i < currToppings.length; i++){
//            System.out.print("Topping - " + currToppings[i].getToppings() + ". Price per Servings $" + currToppings[i].getPriceforLarge()
//            + ".\n");
        }

    public void add(Toppings topping){
        toppings.add(topping);
        this.pizzaTotal = topping.getToppingPrice() + pizzaTotal;
    }

    public double getPizzaTotal() {return pizzaTotal;}
    public Customer getCustomer() {
        return customer;
    }
    public void setPizzaTotal(double pizzaTotal) {
        this.pizzaTotal = pizzaTotal;
    }
    public void setCustomer(Customer customer) {
        this.customer = customer;
    }
}

File: src/PizzaPlus/Pizza.kt
package PizzaPlus

import PizzaPlus.Customer.*
import kotlin.jvm.JvmStatic
import PizzaPlus.Customer
import PizzaPlus.Order
import PizzaPlus.Pizza
import PizzaPlus.Toppings
import java.io.ObjectOutputStream
import java.io.FileOutputStream
import java.io.FileNotFoundException
import java.io.IOException
import java.io.ObjectInputStream
import java.io.FileInputStream
import java.lang.ClassNotFoundException
import kotlin.Throws
import PizzaPlus.XmlHandler
import java.io.File
import java.io.Serializable
import java.util.ArrayList
import javax.xml.parsers.SAXParserFactory

class Pizza(var basePrice: Double) : Serializable {
    var toppings = ArrayList<Toppings>()
    var pizzaTotal = 0.0
    var customer: Customer? = null
    var currToppings: Array<Toppings>? = null
    var toppingsXml = XmlReader("toppings.xml")
    fun toppingsScroller() {
        currToppings = toppingsXml.toppingsList
        for (i in currToppings!!.indices) {}}  // displays topping, need to fix code

    fun add(topping: Toppings) {
        toppings.add(topping)
        pizzaTotal += topping.toppingPrice
    }

//    companion object {
//        var pie12 = 12.00
//        var pie14 = 14.00
//        var pie16 = 16.00
//    }
}